### PR TITLE
Added support for different plurals in different target locales

### PR DESCRIPTION
**Pull request by ehoogerbeets**
- retrieve lists of plurals per language from cldr
- fixed extraction bug in Android resource file
- before adding resources to a new strings file, make sure each
  plural contains the right target plural strings for the
  target locale

### CODE REVIEW

File: generate/genPluralCategories.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+for (var i = 0; i < languages.length; i++ ) {
+    var language = languages[i];
+    var rules = Object.keys(plurals[language]);
+    categories[language] = rules.map(function(rule) {
+        // pluralRule-count-X -> X
+        return rule.substring(17);
+    });
+}
+
+fs.writeFileSync("./db/pluralCategories.json", JSON.stringify(categories, undefined, 4), "utf-8");
```
Comment (gouniLee) [inline generate/genPluralCategories.js:40]: it needs to handle an exceptional case when ./db directory doesn't exist

File: test/testProject.js [hunk 2]
```diff
@@ -125,6 +138,102 @@ module.exports.project = {
         test.done();
     },
     
+    testProjectGeneratesCorrectPluralCategoriesInNewStringsXliffs: function(test){
+        test.expect(9);
```
Comment (gouniLee) [inline test/testProject.js:142]: This feature is only available xliff 1.2 version. isn't it? If so, I think it needs to write the exact xliff working version in the Release note.  or If it's work xliff 2.0 too, test case should be added for xliff 2.0

File: generate/genPluralCategories.js (hunk 1 above)
Comment (ehoogerbeets) [reply generate/genPluralCategories.js:40]: Other things are checked in to the db directory in git, so it is unlikely to not exist

File: test/testProject.js (hunk 2 above)
Comment (ehoogerbeets) [reply test/testProject.js:142]: Added test case for xliff 2.0

**Review by gouniLee — APPROVED** [review]

**Pull request merged**